src/code/firebase/logger.ts

```diff
@@ -0,0 +1,35 @@
+import { GetWasabeeServer } from "../config";
+import { getJWT } from "../auth";
+
+import { port } from "./init";
+
+interface FirebaseMessage {
+  id: string;
+  method: string;
+  action: string;
+  error: string;
+  app_name: string;
+  app_version: string;
+  server: string;
+  jwt: string;
+}
+
+export function postToFirebase(message: Partial<FirebaseMessage>) {
+  // prevent analytics data from being sent if not enabled by the user: GPDR
+  if (
+    message.id == "analytics" &&
+    localStorage[window.plugin.wasabee.static.constants.SEND_ANALYTICS_KEY] !=
+      "true"
+  )
+    return;
+
+  if (message.id == "wasabeeLogin") {
+    message.server = GetWasabeeServer();
+    message.jwt = getJWT();
+  }
+
+  message.app_name = "Wasabee-IITC";
+  message.app_version = window.plugin.wasabee.info.version;
+
+  port.postMessage(message);
```

Comment:
port? or post? I'll have to read the rest of the code...

Reply:
that's port from `import { port } from "./init";`